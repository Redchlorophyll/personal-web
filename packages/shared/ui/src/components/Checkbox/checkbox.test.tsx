import React, { useState } from 'react';
import '@testing-library/jest-dom';
import { Checkbox } from './index';
import {
  renderWithProviders,
  screen,
  userEvent,
  act,
} from 'testing-library-react-custom';

function CheckboxWrapper() {
  const [valueCheckbox, setValueCheckbox] = useState<Array<string>>([]);

  function onChangeCheckbox(val: Array<string>) {
    setValueCheckbox(val);
  }

  return (
    <div>
      {valueCheckbox}
      <Checkbox
        valueList={valueCheckbox}
        onChange={(val: Array<string>) => onChangeCheckbox(val)}
        value={'test'}
      />
    </div>
  );
}

describe('shared - ui - checkbox', () => {
  test('should render properly', () => {
    const { baseElement } = renderWithProviders(<CheckboxWrapper />);

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  test('it should render properly', () => {
    renderWithProviders(<CheckboxWrapper />);
    const target = screen.getByRole('checkbox');
    expect(target).toBeInTheDocument();
  });

  test('it should show value when clicked', async () => {
    const user = userEvent.setup();
    renderWithProviders(<CheckboxWrapper />);
    const target = screen.getByRole('checkbox');
    expect(screen.queryByText('test')).not.toBeInTheDocument();

    await act(async () => await user.click(target));

    expect(screen.queryByText('test')).toBeInTheDocument();

    await act(async () => await user.click(target));
    expect(screen.queryByText('test')).not.toBeInTheDocument();
  });
});
