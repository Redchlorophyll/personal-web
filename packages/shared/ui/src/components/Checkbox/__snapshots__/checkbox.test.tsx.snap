// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared - ui - checkbox should render properly 1`] = `
<body>
  <div>
    <div>
      <label>
        <input
          class="peer w-[24px] h-[24px] sr-only"
          type="checkbox"
          value="test"
        />
        <div
          class="inline-block w-[24px] h-[24px] bg-black-100 border-solid border-[1px] border-black-800 rounded-[3px] peer-checked:bg-primary-800 peer-checked:border-primary-800 p-[6px_3px_4px_1px] peer-checked:drop-shadow-[0px_1px_4px_#6F9DF8]"
        >
          <svg
            fill="none"
            height="11"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m0 4.228 1.29-1.11L7.116 9.89 5.826 11z"
              fill="#fff"
            />
            <path
              d="m5.826 11-1.11-1.29L16.003 0l1.11 1.29z"
              fill="#fff"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</body>
`;
